performance-test/mozlog-neqo-cwnd.py

```diff
@@ -0,0 +1,181 @@
+        "bg_state": [],
+        "bg_time": [],
+
+        # event occurences
+        "p": {}, # pn -> y-axis (bytes in flight after packet sent)
+        "ps": defaultdict(lambda: defaultdict(lambda: [])), # x/y coords of packet_sent/_acked/_lost
+    })
+
+    for line in open(sys.argv[1]):
+        if (result := acked.search(line)) is not None:
+            this = result.group(1)
+            now = get_time(line)
+            data[this]["time"].append(now)
+            data[this]["limited"] = bool(int(result.group(2)))
+            data[this]["last_bytes_in_flight"] = int(result.group(3))
+            data[this]["bif"].append(data[this]["last_bytes_in_flight"])
+            data[this]["cwnd"].append(int(result.group(4)))
+            state = result.group(5)
+            if data[this]["last_state"][0] != state:
+                data[this]["bg_state"].append(state)
+                data[this]["bg_time"].append(now)
+            data[this]["last_state"] = (state, now)
+            data[this]["new_acked"] = result.group(6)
+            if data[this]["limited"]:
+                data[this]["bif_limited"].append(data[this]["last_bytes_in_flight"])
+                data[this]["bif_limited_time"].append(now)
+        elif (result := events.search(line)) is not None:
+            this = result.group(2)
+            now = get_time(line)
+            event = result.group(1)
+            pn = int(result.group(3))
+            packet_size = int(result.group(4))
+            if event == "packet_sent" or event == "packet_acked" or event == "packet_lost":
+                if event == 'packet_sent':
+                    data[this]["last_bytes_in_flight"] += packet_size
+                    data[this]["p"][pn] = data[this]["last_bytes_in_flight"]
+                    if data[this]["last_state"][0] == 'RecoveryStart':
+                        data[this]["bg_state"].append('CongestionAvoidance')
+                        data[this]["bg_time"].append(now)
+                        data[this]["last_state"] = ('CongestionAvoidance', now)
+                    if data[this]["last_state"] == 'PersistentCongestion':
+                        data[this]["bg_state"].append('SlowStart')
+                        data[this]["bg_time"].append(now)
+                        data[this]["last_state"] = ('SlowStart', now)
+                # only remember events for packets where we sent the packet
+                if pn in data[this]["p"]:
+                    data[this]["ps"][event]["time"].append(now)
+                    data[this]["ps"][event]["bif"].append(data[this]["p"][pn])
+                    data[this]["ps"][event]["pn"].append(pn)
+        elif (result := lost.search(line)) is not None:
+            this = result.group(1)
+            now = get_time(line)
+            data[this]["time"].append(now)
+            data[this]["last_bytes_in_flight"] = int(result.group(3))
+            data[this]["bif"].append(last_bytes_in_flight)
```

Comment:
pylance warning (you might want to install the extension if you use VS Code):
```json
[{
	"resource": "/Users/lars/Documents/Code/neqo/performance-test/mozlog-neqo-cwnd.py",
	"owner": "_generated_diagnostic_collection_name_#9",
	"code": {
		"value": "reportUndefinedVariable",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pyright/blob/main/docs/configuration.md",
			"scheme": "https",
			"authority": "github.com",
			"fragment": "reportUndefinedVariable"
		}
	},
	"severity": 4,
	"message": "\"last_bytes_in_flight\" is not defined",
	"source": "Pylance",
	"startLineNumber": 107,
	"startColumn": 38,
	"endLineNumber": 107,
	"endColumn": 58
}]
```

Reply:
This was a dead code path fixed the regex.